src/engine/console.h

```
@@ -70,6 +70,7 @@ class IConsole : public IInterface
 		// DDRace
 
 		virtual int GetVictim() const = 0;
+		bool m_IsAll = false;
```

Could you do the same PR without this? I would like to not introduce command special casing for the `all` parameter, IMO it should always behave the same as if you just passed each tee individually.

---

Why should it behave the same? If I specify an id I want to see all errors if the id is not even connected or similar. If I use the all command I do not want to see 63 errors if I am alone on the server.

---

This is poking from the wrong abstraction level, I think. `IResult` is a nice abstraction, the command should ideally not get told that it's being called with `all`.

Perhaps you could tackle it from the other side and only let the console call the command for ingame players?

---

> Perhaps you could tackle it from the other side and only let the console call the command for ingame players?

I do not understand what that means. What should happen when i run ``kick 22`` and that player is not online? I want an error saying the player is not online. And you are suggesting what exactly in that case?

---

Oh you mean in console.cpp the callback should not be run if not in game. I see. Yes that might work for kicking. But I think in general the m_IsAll is useful to handle also cases where in game players might throw too many errors.

I think it is weird that a command does not know if the is one victim or multiple victims.

---

Checking client state from the console also seems like the wrong abstraction level. And does not cover excluding the person who ran the command. Something like ``unfreeze all`` should apply to all and ``kick all`` should exclude the person who ran it.

---

> Checking client state from the console also seems like the wrong abstraction level. And does not cover excluding the person who ran the command. Something like `unfreeze all` should apply to all and `kick all` should exclude the person who ran it.

I think it'd make sense to do it there, if via a callback that is passed in. In the end, "all" is some kind of wildcard, and it's also the console/shell evaluating the wildcards on Linux, not the individual commands.

---

Even if I somehow technically manage to check client connected in console.cpp I would not know how to check if skipping ``all`` for non connected clients breaks any other command. It sounds reasonablish to do that and it might improve other commands that use victim.

Then we could go for ``unfreeze all`` unfreezing all tees. And ``kick all`` always throwing the error "you can not kick your self" which is meh. But would work. But having 60 errors for non connected tees when doing ``kick all`` on a server with 3 players is a mess. Also having a silent error when providing the wrong id to kick like ``kick 12`` would be bad.